Detour/Source/DetourNode.cpp

```diff
@@ -130,24 +130,36 @@ dtNode* dtNodePool::getNode(dtPolyRef id, unsigned char state)
 		i = m_next[i];
 	}
 	
-	if (m_nodeCount >= m_maxNodes)
+	node = allocNode();
+	if (!node)
 		return 0;
+
+	node->id = id;
+	node->state = state;
+	i = getNodeIdx(node) - 1;
+	m_next[i] = m_first[bucket];
+	m_first[bucket] = i;
 	
-	i = (dtNodeIndex)m_nodeCount;
+	return node;
+}
+
+dtNode* dtNodePool::allocNode()
```

Comment:
I think it would be better to let this take `id` and `state` so we can't allocate nodes and forget to set these parts.